### ISSUE TITLE
pip is deprecating the use of the --global-options install option. 

### ISSUE THREAD

**Issue by Zane-Larking** [pygraphviz/pygraphviz#454]
As of pip-23.1 the installation instructions in INSTALL.txt do not work (at least for windows 11).
Following the steps outlined in [the install instructions](https://pygraphviz.github.io/documentation/stable/install.html#id1) will log:

```
WARNING: Implying --no-binary=:all: due to the presence of --build-option / --global-option / --install-option. Consider using --config-settings for more flexibility.
DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
Collecting pygraphviz
  Using cached pygraphviz-1.10.zip (120 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: pygraphviz
  WARNING: Ignoring --global-option when building pygraphviz using PEP 517
  Building wheel for pygraphviz (pyproject.toml) ... error
  error: subprocess-exited-with-error
```

The work around I've found is to downgrade to pip-23.0
>py -m pip install pip==23.0

**Comment by jeanclaude-jardim-oxb** [comment]
Rather than downgrading pip you can install with `--config-setting`
```
python3 -m pip install \
                --config-setting="--global-option=-I$(brew --prefix graphviz)/include/" \
                --config-setting="--global-option=-L$(brew --prefix graphviz)/lib/" \
                pygraphviz
```

**Comment by rossbar** [comment]
Thanks @jeanclaude-jardim-oxb - if you'd be willing to submit a PR to update the install instructions that'd be great. If you don't have the bandwidth no worries, someone will get around to it eventually. Either way, thanks for the fix!

**Comment by KrisBigK** [comment]
The downgrading worked for me. With pip 23.1 the notorious cgraph.h error haunted me for a couple of hours, and after downgrading to 22.2.2 the installation worked fine.

Before downgrading:
![error 5 fail to install pygraphviz](https://user-images.githubusercontent.com/34199560/233888569-aa00d53c-7319-4f4c-be7a-b0c9fda68ebe.PNG)


After downgrading:
![success 6](https://user-images.githubusercontent.com/34199560/233888623-8b670055-7610-498e-b45c-9d23878de1b1.PNG)

```
PS C:\Users\hlwtg4_Admin> pip install pip==22.2.2
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting pip==22.2.2
  Downloading https://pypi.tuna.tsinghua.edu.cn/packages/1f/2c/d9626f045e7b49a6225c6b09257861f24da78f4e5f23af2ddbdf852c99b8/pip-22.2.2-py3-none-any.whl (2.0 MB)
     ---------------------------------------- 2.0/2.0 MB 7.2 MB/s eta 0:00:00
ERROR: To modify pip, please run the following command:
C:\Users\hlwtg4_Admin\AppData\Local\Programs\Python\Python310\python.exe -m pip install pip==22.2.2

[notice] A new release of pip is available: 23.1 -> 23.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip
PS C:\Users\hlwtg4_Admin> python.exe -m pip install pip==22.2.2
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting pip==22.2.2
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1f/2c/d9626f045e7b49a6225c6b09257861f24da78f4e5f23af2ddbdf852c99b8/pip-22.2.2-py3-none-any.whl (2.0 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 23.1
    Uninstalling pip-23.1:
      Successfully uninstalled pip-23.1
Successfully installed pip-22.2.2
PS C:\Users\hlwtg4_Admin> python -m pip install --global-option=build_ext `
>> --global-option="-IC:\Program Files\Graphviz\include" `
>> --global-option="-LC:\Program Files\Graphviz\lib" `
>> pygraphviz
WARNING: Disabling all use of wheels due to the use of --build-option / --global-option / --install-option.
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting pygraphviz
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/ee/7e/7366c082f959db7ee18a16fc38dc594158ede65ca789bef87751ed5130c7/pygraphviz-1.10.zip (120 kB)
  Preparing metadata (setup.py) ... done
Skipping wheel build for pygraphviz, due to binaries being disabled for it.
Installing collected packages: pygraphviz
  Running setup.py install for pygraphviz ... done
Successfully installed pygraphviz-1.10

[notice] A new release of pip available: 22.2.2 -> 23.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip
```

**Comment by rossbar** [comment]
Closed by #455

**Comment by gotofritz** [comment]
> Rather than downgrading pip you can install with `--config-setting`
> 
> ```
> python3 -m pip install \
>                 --config-setting="--global-option=-I$(brew --prefix graphviz)/include/" \
>                 --config-setting="--global-option=-L$(brew --prefix graphviz)/lib/" \
>                 pygraphviz
> ```

@jeanclaude-jardim-oxb that doesn't work anymore - `error: option -I not recognized`
